axum/src/handler/mod.rs

```diff
@@ -99,6 +99,12 @@ pub(crate) use self::{
 /// {}
 /// ```
 #[doc = include_str!("../docs/debugging_handler_type_errors.md")]
+#[cfg_attr(
+    feature = "nightly-error-messages",
+    rustc_on_unimplemented(
+        note = "Consider using `#[axum_macros::debug_handler]` to improve the error message"
```

Comment:
Nice! 😄 

Can you maybe add one or two tests for the `on_unimplemented` errors?

Reply:
The question is where and how such tests can be added? Running them would always require a nightly compiler + trybuild/compiletests. I'm not that familiar with the axum project setup to answer this question, but if someone points me into the right direction I'm happy to add them there.

Reply:
Although if you're doing this anyways, it's probably more sensible to always compile the test itself, and ignore it on other compiler versions using `#[rustversion::attr(not(nightly), ignore)]`. (instead of having the test do nothing)

Reply:
@weiznich We can also merge this as-is and add those tests in a separate PR, if you prefer?

Reply:
That would be a good idea. I'm currently quite busy with other stuff, so I cannot address this comment now (although I've planed to do that next week or the week after).

Reply:
Okay. Looks like CI needs to be updated though (we're using `--all-features` which is now trying to activate the nightly-only stuff on stable / beta).

Reply:
I'm also fine with making the required changes later this month.

Reply:
I've pushed a new version that enables this by default on nightly compilers and removes the explicit feature flag. Also this is now integrated in the `axum-macros` test suite output, although it's not clear for me if that's run on CI yet?